### ISSUE TITLE
Enable "--display-error-details" in webpack

### ISSUE THREAD

**Issue by munerf** [storybookjs/storybook#6596]
**Is your feature request related to a problem? Please describe.**
To debug module imports I would like to enable webpack's "--display-error-details" inside storybook!

**Describe the solution you'd like**
How can I do it?

**Comment by ndelangen** [comment]
@shilman added a `--debug-webpack` options recently, perhaps we can add this setting when that is active?

**Comment by ndelangen** [comment]
If this flag is toggle-able from the webpack config, this should be fairly simple to implement, any takers?

**Comment by jamieshark** [comment]
Hi there! I think you can enable this webpack setting through the [`stats`](https://webpack.js.org/configuration/stats/) property in a custom config. Based on the [default webpack config](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/configurations/custom-webpack-config/index.md), it doesn't look like there are any default settings for `stats`, so we can assign that as a property on the default config and return it as a new custom config.

1. Create a `.storybook/webpack.config.js` file.
2. Add the `stats` option in your webpack config.
```
// .storybook/webpack.config.js
module.exports = async ({ config, mode }) => {
  config.stats = {
    errorDetails: true, // --display-error-details
  }
  return config;
};
```

Let me know if this works for your use case!

**Comment by Desmond-Campbell** [comment]
j

**Comment by dhuang612** [comment]
Hi,

I would like to take this on!

**Comment by dhuang612** [comment]
Hello, 
created PR #8347

**Comment by dhuang612** [comment]
Hi, 

I am pretty new to webpack. I've done some research into how to setup this via stats
https://webpack.js.org/configuration/stats/#statserrordetails

The files I see related to Webpack in this project are under
lib > core > src > server > preview
with the two files being: 
**base-webpack.config.js** and ** custom-webpack-preset.js**
But then I also see a dev-server.js file.

Could I get some help on how I'm supposed to approach this?

Thanks!

**Comment by sakito21** [comment]
@shilman 
This fixed #8391  Issue. 
I think you can close it.
